src/Core/AbstractRequest.php

```diff
@@ -8,11 +8,20 @@
      */
     protected $client;
 
+    protected $asJson = false;
+
     /**
      * @param AbstractClient $client
      */
     public function __construct(AbstractClient $client)
     {
         $this->client = $client;
     }
+
+    public function asJson()
```

Comment:
Adding it this way is because of the chaining functionallity.. I think this looks nicer. 

But yeah it's not that pretty.